histomicstk/features/ComputeFSDs.py

```diff
@@ -0,0 +1,179 @@
+import collections
+import numpy as np
+from skimage.segmentation import find_boundaries
+
+
+def ComputeFSDs(Label, K=128, Fs=6):
+    """
+    Calculates `Fourier shape descriptors` from a label mask.
+
+    Parameters
+    ----------
+    Label : array_like
+        A M x N label mask.
+    K : Number of points for boundary resampling to calculate fourier
+        descriptors. Default value = 128.
+    Fs : Number of frequency bins for calculating FSDs. Default value = 6.
+
+    Returns
+    -------
+    FSDGroup : array_like
+        A 1 x Fs FSD arrays.
+
+    References
+    ----------
+    .. [1] D. Zhang et al. "A comparative study on shape retrieval using
+       Fourier descriptors with different shape signatures," In Proc.
+       ICIMADE01, 2001.
+    """
+
+    # initialize FSD feature group
+    FSDGroup = np.zeros(Fs)
+
+    # find nucleus boundaries
+    Bounds = np.argwhere(
+        find_boundaries(Label, mode="inner").astype(np.uint8) == 1
+    )
+
+    # fourier descriptors, spaced evenly over the interval 1:K/2
+    Interval = np.round(
+        np.power(
+            2, np.linspace(0, np.log2(K)-1, Fs+1, endpoint=True)
+        )
+    ).astype(np.uint8)
+
+    FSDGroup = FSDs(Bounds[:, 0], Bounds[:, 1], K, Interval)
+
+    return FSDGroup
+
+
+def InterpolateArcLength(X, Y, L):
```

Comment:
I assume we will catch all these private functions when we re-name to snake_case. No harm in doing some now though as you encounter them.